core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -906,7 +906,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       val results = offsetFetchRequest.partitions.asScala.map { topicPartition => (topicPartition, unauthorizedGroupResponse)}.toMap
       new OffsetFetchResponse(results.asJava)
     } else {
-      val (authorizedTopicPartitions, unauthorizedTopicPartitions) = offsetFetchRequest.partitions.asScala.partition { topicPartition =>
+      val partitions =
+        if (header.apiVersion == 2 && offsetFetchRequest.isAllPartitions()) {
+          val group = coordinator.groupManager.getGroup(offsetFetchRequest.groupId)
```

Comment:
Hmm... I don't think we should be accessing the group directly at this layer. It's probably better to either overload `GroupCoordinator.handleFetchOffsets` or expose a new method.